src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -536,12 +557,34 @@ private async Task<string> GetGistInternal(string gistId, RuntimeContext<TResour
             return invoker.EntityMetadata.ScriptText;
         }
 
-        private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(RuntimeContext<TResource> context)
+        private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(RuntimeContext<TResource> context, string queryText=null)
         {
             await this._sourceWatcherService.Watcher.WaitForFirstCompletion();
+            var allDetectors = _invokerCache.GetEntityInvokerList<TResource>(context).ToList();
+
+            SearchResults searchResults = null;
+            if (queryText != null && queryText.Length>3)
```

Comment:
`queryText.Length > 3` - Can I know why there is minimum length restriction of 3 above? What would happen if its less than 3?

Reply:
A query of length <= 3 does not make much sense as such words are mostly stopwords for our search. If the size is 3 or less we will treat it as no query i.e. we would return the list of all detectors for that resource as we would otherwise do.